### PR TITLE
v0.12.0: sources, unlimited versions, and all deployments

### PR DESCRIPTION
**Pull request by pirapira**
Sometimes it's hard to find the Solidity source corresponding to a particular deployment. https://github.com/raiden-network/raiden-contracts/issues/556  This PR makes the process easier by copying the sources to `data` directory whenever they are compiled.

With this change, the precompiled bytecode is always accompanied by the source it's compiled from.

### CODE REVIEW

**Comment by pirapira** [comment]
Review #617 first.

**Comment by pirapira** [comment]
@rakanalh I tried `sdist` and I see the sources in `data*/` directories.
```
% rm -rf dist
% python setup.py sdist
% cd dist
% tar xf raiden-contracts-0.12.0.tar.gz
% cd raiden-contracts-0.12.0
% ls -d raiden_contracts/*/source
raiden_contracts/data_0.8.0/source            raiden_contracts/data/source
raiden_contracts/data_0.8.0_unlimited/source  raiden_contracts/data_unlimited/source
```

File: raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+        deprecation_executor = msg.sender;
+    }
+
+    /// @notice Deploy a new TokenNetwork contract for the Token deployed at
+    /// `_token_address`.
+    /// @param _token_address Ethereum address of an already deployed token, to
+    /// be used in the new TokenNetwork contract.
+    function createERC20TokenNetwork(address _token_address)
+         canCreateTokenNetwork
```
Comment (rakanalh) [inline raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol]: Nickpick: Indentation of `canCreateTokenNetwork` is different than the `external` below it

**Comment by pirapira** [comment]
> does it make sense to have a directory called `data` which contains all the sources & json files for that version as well as `data_0.8.0`

Yes, that simplifies the release process.

According to your idea, when I release 0.10.0, I have to find an old version of `data` and copy that into `data_0.9.0`.  When I release 0.10.0, I don't want to think about 0.9.0.

File: raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol (hunk 1 above)
Comment (pirapira) [reply raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol]: I'll fix this in the PR.

Comment (pirapira) [reply raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol]: No I don't.  Then I have to deploy contracts 21 times again.

Comment (pirapira) [reply raiden_contracts/data/data_0.8.0/source/raiden/TokenNetworkRegistry.sol]: I created an issue about that https://github.com/raiden-network/raiden-contracts/issues/626 .

**Comment by pirapira** [comment]
I don't even know whether the PyPI build succeeds.  I'm more interested in that, rather than the number of identical source files.

**Pull request merged**